shared/middleware/files_api.rb

```diff
@@ -206,12 +238,44 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
     put_file('assets', encrypted_channel_id, file[:filename], file[:tempfile].read)
   end
 
+  # POST /v3/animations/<channel-id>/<filename>?version=<version-id>
+  #
+  # Create or replace an animation. We use this method so that IE9 can still
+  # upload by posting to an iframe.
+  #
+  post %r{/v3/(animations)/([^/]+)/([^/]+)$} do |endpoint, encrypted_channel_id, filename|
+    dont_cache
+    # though this is JSON data, we're making the POST request via iframe
+    # form submission. IE9 will try to download the response if we have
+    # content_type json
+    content_type 'text/plain'
+
+    bad_request unless request.POST['files'] && request.POST['files'][0]
+
+    file = request.POST['files'][0]
+
+    bad_request unless file[:filename] && file[:tempfile]
+
+    put_file(endpoint, encrypted_channel_id, filename, file[:tempfile].read)
+  end
+
+  # POST /v3/animations/<channel-id>/<filename>/from/<source-filename>
+  #
+  # Create or replace an animation. We use this method so that IE9 can still
+  # upload by posting to an iframe.
+  #
+  post %r{/v3/(animations)/([^/]+)/([^/]+)/from/([^/]+)$} do |endpoint, encrypted_channel_id, filename, source_filename|
```

Comment:
Are these copies going to be coming from the jquery file uploader plugin? If not, I would recommend using PUT instead of POST. 

Also, I would recommend using a query param ?src=<scrname> instead of /from/<srcname> since it seems more RESTful: http://stackoverflow.com/questions/18755220/what-is-the-restful-way-to-represent-a-resource-clone-operation-in-the-url (note that the PUT/POST situation is different in this article)